src/contexts/ApiPromiseContextProvider.tsx

```diff
@@ -33,25 +33,32 @@ export function ApiPromiseContextProvider(
 		new ApiPromise({ provider, types })
 	);
 	const [isReady, setIsReady] = useState(false);
+
 	useEffect(() => {
 		if (isReady) {
 			setIsReady(false);
+			apiPromise.disconnect().then( () => console.log(`${contextType} Resetting connection` ));
 		}
+	// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can you elaborate why this is needed?

Reply:
Because i just want the effect running only when sourceTarget is changed, otherwise that eslint rule is complaining because i'm not putting as a dependency apiPromise, contextType etc... which i don't want to.
Actually i don't see the benefits of that eslint rule which comes with create react app to be honest